typing/typemod.ml

```diff
@@ -2097,22 +2097,31 @@ let wrap_constraint_with_shape env mark arg mty
 
 
 (* Summary for F(X) *)
-type application_summary = {
-  loc: Location.t;
-  attributes: attributes;
-  f_loc: Location.t; (* loc for F *)
-  arg_is_syntactic_unit: bool;
-  arg: Typedtree.module_expr;
-  arg_path: Path.t option;
-  shape: Shape.t
-}
-
-let simplify_app_summary app_view =
-  let mty = app_view.arg.mod_type in
-  match app_view.arg_is_syntactic_unit , app_view.arg_path with
-  | true,   _ -> Includemod.Error.Unit, mty
-  | false, Some p -> Includemod.Error.Named p, mty
-  | false, None -> Includemod.Error.Anonymous, mty
+type application_summary =
+  | Normal_app of {
+      loc: Location.t;
+      attributes: attributes;
+      f_loc: Location.t; (* loc for F *)
+      arg_is_syntactic_unit: bool;
+      arg: Typedtree.module_expr;
+      arg_path: Path.t option;
+      shape: Shape.t;
+    }
+  | Unit_app of {
+      loc: Location.t;
+      attributes: attributes;
+      f_loc: Location.t; (* loc for F *)
+    }
+
+let simplify_app_summary = function
+  | Unit_app _ ->
+    Includemod.Error.Unit, Mty_signature []
```

Comment:
Note: this is not something your PR should fix, but here as well there is a mismatch where we handle `()` as if its signature was `sig end`, instead of having an explicit representation for its signature `()`.